rapids-4-spark-tools/README.md

```diff
@@ -1,78 +1,197 @@
+
+Here are 2 options:
+1. Download the jar file from maven repository
+
+2. Compile the jar from github repo
+```bash
+git clone https://github.com/NVIDIA/spark-rapids.git
+cd spark-rapids
+mvn -pl .,rapids-4-spark-tools clean verify -DskipTests
+```
+The jar is generated in below directory :
+
+`./rapids-4-spark-tools/target/rapids-4-spark-tools_2.12-<version>.jar`
+
+## How to use this tool
+
+This tool parses the spark event log(s) and creates an output report.
+Acceptable input spark event log paths are local filesystem, hdfs , S3 or mixed.
+It can be file path , directory path or mixed.
+
+Below is an example input:
+```
+/path/to/localfilesystem/eventlog1 hdfs://path/to/directory/ s3a://<BUCKET>/to/directory/
+```
+
+If any input is a S3 file path or directory path, here 2 extra steps to access S3 in Spark:
+1. Download the matched jars based on the Hadoop version:
+   - hadoop-aws-<version>.jar
+   - aws-java-sdk-<version>.jar 
+     
+Take Hadoop 2.7.4 for example, we can download and include below jars in the '--jars' option to spark-shell or spark-submit:
+[hadoop-aws-2.7.4.jar](https://repo.maven.apache.org/maven2/org/apache/hadoop/hadoop-aws/2.7.4/hadoop-aws-2.7.4.jar) and 
+[aws-java-sdk-1.7.4.jar](https://repo.maven.apache.org/maven2/com/amazonaws/aws-java-sdk/1.7.4/aws-java-sdk-1.7.4.jar)
+
+2. In $SPARK_HOME/conf, create `hdfs-site.xml` with below AWS S3 keys inside:
+```xml
+<?xml version="1.0"?>
+<configuration>
+<property>
+  <name>fs.s3a.access.key</name>
+  <value>xxx</value>
+</property>
+<property>
+  <name>fs.s3a.secret.key</name>
+  <value>xxx</value>
+</property>
+</configuration>
+```
+
+It has 2 functions -- qualification and profiling.
 
 ## Qualification Tool
 
-### Use from Spark
+### Use from spark-shell
 1. Include rapids-4-spark-tools_2.12-<version>.jar in the '--jars' option to spark-shell or spark-submit
 2. After starting spark-shell:
 
 For multiple event logs comparison and analysis:
-```
+```bash
 com.nvidia.spark.rapids.tool.qualification.QualificationMain.main(Array("/path/to/eventlog1", "/path/to/eventlog2"))
 ```
 
-### How to compile and use from command-line
-1. `./bin/spark-submit --class com.nvidia.spark.rapids.tool.qualification.QualificationMain <Spark-Rapids-Repo>/rapids-4-spark-tools/target/rapids-4-spark-tools_2.12-<version>.jar /path/to/eventlog1 /path/to/eventlog2`
+### Use from spark-submit
+```bash
+$SPARK_HOME/bin/spark-submit --class com.nvidia.spark.rapids.tool.qualification.QualificationMain \
+<Spark-Rapids-Repo>/rapids-4-spark-tools/target/rapids-4-spark-tools_2.12-<version>.jar \
+/path/to/eventlog1 /path/to/eventlog2
+```
+
+### Options
+```bash
+$SPARK_HOME/bin/spark-submit \
+--class com.nvidia.spark.rapids.tool.qualification.QualificationMain \
+<Spark-Rapids-Repo>/rapids-4-spark-tools/target/rapids-4-spark-tools_2.12-<version>.jar \
+--help
+
+For usage see below:
+
+  -i, --include-exec-cpu-percent   Include the executor CPU time percent. It
```

Comment:
it would be nice to perhaps have a section explaining this in more details as to what it is and also do not run with over 100 applications at a time and use a lot of driver memory (30g)

Reply:
Added a section for this.

Reply:
How is current explanation?

Reply:
its fine for now